### PR TITLE
Http Methods updated and path name make it configurable

### PR DESCRIPTION
**Pull request by sanjeevkumar0412**
API Http Methods updated and path name make it configurable

### CODE REVIEW

File: app/services/database.py [hunk 1]
```diff
@@ -471,7 +473,6 @@ def update_token(self, session,record_id, user_name):
 
     def get_connection_string(self, server, database, client_id):
         try:
-            dns = f'mssql+pyodbc://FLM-VM-COGAIDEV/AudioTrans?driver=ODBC+Driver+17+for+SQL+Server'
```
Comment (ashutoshjoshi111) [inline app/services/database.py:474]: dns = f'mssql+pyodbc://FLM-VM-COGAIDEV/AudioTrans?driver=ODBC+Driver+17+for+SQL+Server'

Can you make it fully configurable?

File: flask_app.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ def get_recordby_id():
     return data
 
 
-@app.route('/get_record_by_column_name', methods=['GET','POST'])
+@app.route('/get_record_by_column_name', methods=['GET'])
 @check_authorization
 def get_recordby_column_name():
     table_name = request.args.get('table_name')
```
Comment (ashutoshjoshi111) [inline flask_app.py]: Do we have null or type checking?

File: flask_app.py [hunk 3]
```diff
@@ -140,7 +140,7 @@ def get_recordby_column_name():
     return data
 
 
-@app.route('/update_record_by_column', methods=['GET','POST'])
+@app.route('/update_record_by_column', methods=['PUT'])
 @check_authorization
 def get_update_by_column_name():
     table_name = request.args.get('table_name')
```
Comment (ashutoshjoshi111) [inline flask_app.py]: Null checking or type checking?

File: flask_app.py [hunk 4]
```diff
@@ -194,7 +194,7 @@ def delete_recordby_id():
     return {'data': data}
 
 
-@app.route('/merge_chunk_transcribe_text', methods=['GET','POST'])
+@app.route('/merge_chunk_transcribe_text', methods=['GET'])
 @check_authorization
 def get_transcribe_sentiment():
     client_id_val = request.args.get('clientid')
```
Comment (ashutoshjoshi111) [inline flask_app.py]: Null or type checking?

File: flask_app.py [hunk 5]
```diff
@@ -215,7 +215,7 @@ def get_transcribe_sentiment():
     return data
 
 
-@app.route('/get_client_master_configurations', methods=['GET','POST'])
+@app.route('/get_client_master_configurations', methods=['GET'])
 def get_client_master_configurations():
     # Done
     client_id_val = request.args.get('clientid')
```
Comment (ashutoshjoshi111) [inline flask_app.py]: Null or type checking?

File: flask_app.py [hunk 6]
```diff
@@ -548,7 +545,7 @@ def update_token():
     result = update_authentication_token(server_name, database_name, client_id,user_name)
```
Comment (ashutoshjoshi111) [inline flask_app.py:543]: Can we try/catch and error handling??

File: flask_app.py [hunk 7]
```diff
@@ -595,7 +592,7 @@ def get_prohibited_data():
     data = sentiment_instance.get_prohibited_data_from_table(server_name, database_name, client_id)
```
Comment (ashutoshjoshi111) [inline flask_app.py:590]: Can we use try catch and error handing?

File: flask_app.py [hunk 8]
```diff
@@ -810,8 +807,8 @@ def get_call_status():
     data = get_job_staus_from_audiotranscribe_table(server_name, database_name, client_id, column_name,column_value, page, per_page,from_date,to_date)
     return data
 
-UPLOAD_FOLDER = 'C:/AICogent/ICFiles/'
-ALLOWED_EXTENSIONS = {'wav','mp3'}
+UPLOAD_FOLDER = os.environ.get("Upload_Path")
+ALLOWED_EXTENSIONS = {'wav','mp3'}  #Mark in env
```
Comment (ashutoshjoshi111) [inline flask_app.py:809]: Why is it hardcoded??

File: flask_app.py [hunk 9]
```diff
@@ -846,7 +843,7 @@ def upload_page():
 def uploaded_file(filename):
```
Comment (ashutoshjoshi111) [inline flask_app.py:841]: what's the use of this single line function?

File: flask_app.py [hunk 10]
```diff
@@ -936,7 +933,7 @@ def generate_text():
 
 
 
-@app.route('/client_audio_information', methods=['GET','POST'])
+@app.route('/client_audio_information', methods=['GET'])
 @check_authorization
 def get_client_audio_info():
     client_id_val = request.args.get('clientid')
```
Comment (ashutoshjoshi111) [inline flask_app.py:937]: Error handling, try/catch, logging and other standard conventions?

File: flask_app.py [hunk 11]
```diff
@@ -956,6 +953,43 @@ def get_client_audio_info():
...
+    except Exception as e:
+        response_message = 'You were given the wrong parameter by them. Please try again with a valid parameter.'
+        return {
+            "result": [response_message],
+            "message": response_message,
+            "status": 'failed',
+            'status_code': RESOURCE_NOT_FOUND
+        }, RESOURCE_NOT_FOUND
+    # client_id = int(request.args.get('clientid'))
+    audio_file_name = request.args.get('audio_file')
```
Comment (ashutoshjoshi111) [inline flask_app.py:970]: type checking, Null and blank handling, error log and other standard practices.

File: flask_app.py [hunk 12]
```diff
@@ -956,6 +953,43 @@ def get_client_audio_info():
...
+    return data
+# conversational
+
+@app.route('/conversational',methods=['GET'])
+def conversational_data():
+    return render_template('conversational.html')
+
+@app.route('/get_conversation', methods=['POST'])
+def get_conversation():
+    # Get the parameters sent by the Ajax request
```
Comment (ashutoshjoshi111) [inline flask_app.py:982]: Null and error handling, logging etc.

File: flask_app.py [hunk 13]
```diff
@@ -96,7 +96,7 @@ def get_record():
     return data
 
 
-@app.route('/get_record_by_id', methods=['GET','POST'])
+@app.route('/get_record_by_id', methods=['GET'])
 def get_recordby_id():
```
Comment (ashutoshjoshi111) [inline flask_app.py:100]: Please make sure to handle null and error, log all the details.

File: app/services/database.py [hunk 14]
```diff
@@ -107,8 +108,10 @@ def create_audio_file_entry(self, model_info):
         try:
             # db_server = self.global_utility.get_database_server_name()
             # db_name = self.global_utility.get_database_name()
-            db_server = 'FLM-VM-COGAIDEV'
-            db_name = 'AudioTrans'
+            # db_server = 'FLM-VM-COGAIDEV'
+            db_server = os.environ.get("SERVER_NAME")
+            db_name = os.environ.get("DB_NAME")
+            # db_name = 'AudioTrans'
             dns = f'mssql+pyodbc://{db_server}/{db_name}?driver=ODBC+Driver+17+for+SQL+Server'
```
Comment (ashutoshjoshi111) [inline app/services/database.py]: Database string can't be hardcoded partially.

File: .env [hunk 15]
```diff
@@ -9,5 +9,7 @@ DATABASE_NAME_DEV = 'AudioMaster'
 DATABASE_NAME_QA = 'AudioMasterQA'
 open_ai_model = 'gpt-4-1106-preview'
 access_token='No'
-bussiness_model='cogent'
+bussiness_model='No-cogent'
```
Comment (ashutoshjoshi111) [inline .env:12]: is that still hardcoded?

File: app/services/database.py [hunk 16]
```diff
@@ -435,7 +438,6 @@ def generate_token(self, session,client_id, user_name):
             session.commit()
             self.logger.info(f"Record inserted successfully. ID: {record_model.Id}")
             return record_model
-            print("Generated token:", record_model.Id)
```
Comment (ashutoshjoshi111) [inline app/services/database.py:438]: Ok, also remove extra print from rest of the code base.

File: flask_app.py [hunk 17]
```diff
@@ -430,14 +430,11 @@ def match_file_name_pettern():
         }, RESOURCE_NOT_FOUND
     file_name = request.args.get('filename')
     current_user = os.getlogin()
-    file_name = '24-10003-douglas-21March-AY-Noida-Call-Approva-Ashutosh'
-    file_name = '24-10003_tomous_25April_CTS_Mumbai_Outbound_Ritesh_Manish'
-    # file_name = 'ABC-21March-AY-Noida-Call-Approva'
     print('Current login user:', current_user)
```
Comment (ashutoshjoshi111) [inline flask_app.py:431]: Remove extra prints.

File: flask_app.py [hunk 18]
```diff
@@ -355,7 +355,7 @@ def get_sentiment_data():
     return data
 
 
-@app.route('/get_all_configurations', methods=['GET','POST'])
+@app.route('/get_all_configurations', methods=['GET'])
 def get_all_configurations():
```
Comment (ashutoshjoshi111) [inline flask_app.py:357]: Do we have try catch and audit and error logging enabled for this function?

**Review by ashutoshjoshi111 — COMMENTED** [review]
Can you please take a look at the comments, please discuss then if required?

File: flask_app.py (hunk 7 above)
Comment (sanjeevkumar0412) [reply flask_app.py:590]: In the main Function **get_prohibited_data_from_table** already taken care of the error handling that's why not needed here

File: .env (hunk 15 above)
Comment (sanjeevkumar0412) [reply .env:12]: No ..these values in the Env file

File: flask_app.py (hunk 7 above)
Comment (ashutoshjoshi111) [reply flask_app.py:590]: Ok, please also use try catch in the inner function, please record the error and then throw exception to the main function.

File: .env (hunk 15 above)
Comment (ashutoshjoshi111) [reply .env:12]: Excellent!!

File: app/services/database.py (hunk 14 above)
Comment (sanjeevkumar0412) [reply app/services/database.py]: ODBC driver is also configurable reflected on this changeset e1ae681

File: flask_app.py (hunk 17 above)
Comment (sanjeevkumar0412) [reply flask_app.py:431]: code Modified and removed the unwanted print statement reflected on this changeset e1ae681

File: app/services/database.py (hunk 16 above)
Comment (sanjeevkumar0412) [reply app/services/database.py:438]: code Modified and removed the unwanted print statement reflected on this changeset e1ae681

File: flask_app.py (hunk 6 above)
Comment (sanjeevkumar0412) [reply flask_app.py:543]: Internally added the exception handling

File: flask_app.py (hunk 5 above)
Comment (sanjeevkumar0412) [reply flask_app.py]: Added in the exception handling code reflected on this changeset e1ae681

File: flask_app.py (hunk 4 above)
Comment (sanjeevkumar0412) [reply flask_app.py]: Added in the exception handling code reflected on this changeset e1ae681

File: flask_app.py (hunk 3 above)
Comment (sanjeevkumar0412) [reply flask_app.py]: Added in the exception handling code reflected on this changeset

File: flask_app.py (hunk 2 above)
Comment (sanjeevkumar0412) [reply flask_app.py]: Added in the exception handling code reflected on this changeset

File: flask_app.py (hunk 3 above)
Comment (sanjeevkumar0412) [reply flask_app.py]: e1ae681

File: flask_app.py (hunk 2 above)
Comment (sanjeevkumar0412) [reply flask_app.py]: e1ae681

File: app/services/database.py (hunk 1 above)
Comment (sanjeevkumar0412) [reply app/services/database.py:474]: ODBC driver now is configurable on this changeset e1ae681

File: flask_app.py (hunk 18 above)
Comment (sanjeevkumar0412) [reply flask_app.py:357]: yes error logging is there

File: flask_app.py (hunk 7 above)
Comment (sanjeevkumar0412) [reply flask_app.py:590]: Done

**Pull request closed without merging**